app.py

```diff
@@ -21,6 +21,7 @@
 
 app = Flask(__name__)
 babel = Babel(app)
+app.secret_key = '\x10x\x1a\xc4\x0ef6\xbdcy\x0b\xba\xf8KV\x19\x02\x82\xed\x9b;\xca9#'
```

Comment:
This code is public; we can't put a secret key in here. You can pull it from a sitewide configuration setting instead.

This whole login thing looks like something that should be in a separate branch. I would be okay with merging revision cbc38cea00147b9c7c5cea6d59c74626d966b6e4 and addressing the login bit separately.